v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/rowmapper/provider/PostgreSQLJdbcValueMappings.java

```diff
@@ -0,0 +1,71 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.source.reader.io.jdbc.rowmapper.provider;
+
+import com.google.cloud.teleport.v2.source.reader.io.jdbc.rowmapper.JdbcValueMapper;
+import com.google.cloud.teleport.v2.source.reader.io.jdbc.rowmapper.JdbcValueMappingsProvider;
+import com.google.cloud.teleport.v2.source.reader.io.jdbc.rowmapper.ResultSetValueExtractor;
+import com.google.cloud.teleport.v2.source.reader.io.jdbc.rowmapper.ResultSetValueMapper;
+import com.google.cloud.teleport.v2.source.reader.io.schema.typemapping.provider.unified.CustomSchema.Date;
+import com.google.common.collect.ImmutableMap;
+import java.sql.ResultSet;
+import java.util.Calendar;
+import java.util.Map;
+import java.util.TimeZone;
+import org.apache.avro.generic.GenericRecordBuilder;
+import org.apache.commons.lang3.tuple.Pair;
+
+/** PostgreSQL data type mapping to AVRO types. */
+public class PostgreSQLJdbcValueMappings implements JdbcValueMappingsProvider {
+  private static final ResultSetValueMapper<?> valuePassThrough = (value, schema) -> value;
+
+  private static final Calendar utcCalendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+
+  private static final ResultSetValueExtractor<java.sql.Date> utcDateExtractor =
+      (rs, fieldName) -> rs.getDate(fieldName, utcCalendar);
+
+  private static final ResultSetValueMapper<java.sql.Date> sqlDateToAvro =
+      (value, schema) ->
+          new GenericRecordBuilder(Date.SCHEMA)
+              .set(Date.DATE_FIELD_NAME, (int) value.toLocalDate().toEpochDay())
+              .build();
+
+  // TODO(thiagotnunes): Add missing type mappings
+  private static final ImmutableMap<String, JdbcValueMapper<?>> SCHEMA_MAPPINGS =
+      ImmutableMap.<String, Pair<ResultSetValueExtractor<?>, ResultSetValueMapper<?>>>builder()
+          .put("BIGINT", Pair.of(ResultSet::getLong, valuePassThrough))
+          .put("CHARACTER VARYING", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("DATE", Pair.of(utcDateExtractor, sqlDateToAvro))
+          .put("INT8", Pair.of(ResultSet::getLong, valuePassThrough))
+          .put("TEXT", Pair.of(ResultSet::getString, valuePassThrough))
+          .put("VARCHAR", Pair.of(ResultSet::getString, valuePassThrough))
+          .build()
+          .entrySet()
+          .stream()
+          .map(
+              entry ->
+                  Map.entry(
+                      entry.getKey(),
+                      new JdbcValueMapper<>(
+                          entry.getValue().getLeft(), entry.getValue().getRight())))
+          .collect(ImmutableMap.toImmutableMap(Map.Entry::getKey, Map.Entry::getValue));
+
+  /** Get static mapping of SourceColumnType to {@link JdbcValueMapper}. */
+  @Override
+  public ImmutableMap<String, JdbcValueMapper<?>> getMappings() {
+    return SCHEMA_MAPPINGS;
```

Comment:
In case we are taking up UT in a subsequent PR, please add TODOs whereever coverage is missing or the providers and dialect adapter classes.

Reply:
Added TODOs